index.php

```diff
@@ -36,15 +36,16 @@
 if (file_exists($maintenanceFile)) {
     $maintenanceBypass = false;
 
-    if (file_exists($maintenanceIpFile)) {
+    if (file_exists($maintenanceIpFile) && is_readable($maintenanceIpFile)) {
         /* if maintenanceFile and maintenanceIpFile are set use Mage to get remote IP (in order to respect remote_addr_headers xml config) */
         Mage::init($mageRunCode, $mageRunType);
         $currentIp = Mage::helper('core/http')->getRemoteAddr();
-        $allowedIps = explode(',', trim(file_get_contents($maintenanceIpFile)));
-
-        if (in_array($currentIp, $allowedIps)) {
-            /* IP address matches, bypass maintenanceMode */
-            $maintenanceBypass = true;
+        $allowedIps = preg_split('/[\ \n\,]+/', file_get_contents($maintenanceIpFile), 0, PREG_SPLIT_NO_EMPTY);
+        foreach ($allowedIps as $allowedIp) {
+            if ($currentIp === $allowedIp && !filter_var($allowedIp, FILTER_VALIDATE_IP)) {
```

Comment:
```suggestion
            if ($currentIp === $allowedIp && filter_var($allowedIp, FILTER_VALIDATE_IP)) {
```

However, I think the check for valid IP is unnecessary. If `$currentIp === $allowedIp` is true, it means `$allowedIp` must be a valid IP.  So I think the loop can be replaced with `in_array($currentIp, $allowedIps)`.

Reply:
I also think checking for valid IP is overkill, better to keep this code as small and fast as possible IMHO